src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -1384,21 +1366,48 @@ private void EmitUnmarshalFunctionArgs(List<AssociativeNode> args, List<Type> pa
                 }
             }
         }
-
-        private bool DoesParamArgRankMatch(List<Type> parameterTypes, List<AssociativeNode> args)
+        /// <summary>
+        /// Checks that all arguments are compatible with the parameters of a given fep.
+        /// Checks rank, type, and attributes to make this determination.
+        /// </summary>
+        /// <param name="fep"></param>
+        /// <param name="parameterTypes"></param>
+        /// <param name="args"></param>
+        /// <returns></returns>
+        /// <exception cref="Exception"></exception>
+        private bool DoParamsRankMatchArgs(ProtoCore.CLRFunctionEndPoint fep, List<Type> parameterTypes, List<AssociativeNode> args)
         {
+            //TODO this can throw because it's possible that the FEP is incorrect
+            //and if this is the case we should throw or fallback to replication -
+            //in either case log something - we need compile time overload support.
+            if (parameterTypes.Count != args.Count)
```

Comment:
Are the counts for these always expected to be the same irrespective of whether the method is static or instance?

Reply:
this is a good question - 
initially I did not think so, but then regarding this:
https://github.com/DynamoDS/Dynamo/pull/13712#discussion_r1096028816 it seems that the number of args/params are always the same...

I did find this code which seems to imply unifying the number of params:
https://github.com/DynamoDS/Dynamo/blob/emitMSIL-V2/src/Engine/EmitMSIL/CodeGenIL.cs#L417